scripts/did_finder.py

```diff
@@ -26,157 +26,91 @@
-        print("----> ", did_request)
-        servicex = ServiceXAdapter(did_request['service-endpoint'])
-        did = did_request['did']
-        request_id = did_request['request_id']
-
-        servicex.post_status_update("DID Request received")
-        lookup_request = LookupRequest(
-            request_id=request_id,
-            did=did,
-            rucio_adapter=rucio_adapter,
-            servicex_adapter=servicex,
-            site=site,
-            prefix=prefix,
-            chunk_size=1000,
-            threads=threads
-        )
-
-        lookup_request.lookup_files()
-
-    except Exception:
-        traceback.print_exc()
-        exc_type, exc_value, exc_traceback = sys.exc_info()
-        servicex.post_status_update("DID Request failed " + str(exc_value))
-    finally:
-        channel.basic_ack(delivery_tag=method.delivery_tag)
-
-
-def init_rabbit_mq(rabbitmq_url, retries, retry_interval):
-    rabbitmq = None
-    retry_count = 0
-
-    while not rabbitmq:
-        try:
-            rabbitmq = pika.BlockingConnection(pika.URLParameters(rabbitmq_url))
-            _channel = rabbitmq.channel()
-            _channel.queue_declare(queue=QUEUE_NAME)
-
-            print("Connected to RabbitMQ. Ready to start consuming requests")
-
-            _channel.basic_consume(queue=QUEUE_NAME,
-                                   auto_ack=False,
-                                   on_message_callback=callback)
-            _channel.start_consuming()
+from servicex_did_finder_lib import default_command_line_args, start_did_finder
+from servicex.did_finder.lookup_request import LookupRequest
 
-        except pika.exceptions.AMQPConnectionError as eek:
-            rabbitmq = None
-            retry_count += 1
-            if retry_count < retries:
-                print("Failed to connect to RabbitMQ. Waiting before trying again")
-                time.sleep(retry_interval)
-            else:
-                print("Failed to connect to RabbitMQ. Giving Up")
-                raise eek
 
+async def find_files(rucio_adaptor: RucioAdapter, site: str, prefix: str, threads: int,
+                     did_name: str, info: Dict[str, Any]) \
+                     -> AsyncGenerator[Dict[str, Any], None]:
+    '''For each incoming did name, generate a list of files that ServiceX can process
+    from the rucio catalog
+
+    Args:
+        did_name (str): Rucio Dataset name
+        info (Dict[str, Any]): Various bits of info about the request.
+
+    Returns:
+        AsyncGenerator[Dict[str, any], None]: yield each file
+    '''
+
+    lookup_request = LookupRequest(
+        did=did_name,
+        rucio_adapter=rucio_adaptor,
+        site=site,
+        prefix=prefix,
+        chunk_size=1000,
+        threads=threads,
+        request_id=info['request-id']
+    )
+
+    async for f in lookup_request.lookup_files():
+        yield f
+
+
+def run_rucio_finder():
+    '''Run the rucio finder
+    '''
+    logger = logging.getLogger(__name__)
```

Comment:
Shouldn't this be using the logging code from the library?  Seems like we miss the requestId handling here.  Also, I believe you can just use logging.getLogger() here.

Reply:
Not exactly sure what the proper thing to do here is:

1. Python recommends using `logging.getLogger(__name__)`.
2. `logging.getLogger()` gets the root logger, which is what is configured in the did logger library
3. When you do something like `logging.getLogger(__name__).info('bring it on!')`, then it will use the root logger in the end to dump the text in the end, since it just climbs up the hierarchy until it gets to a handler in the root logger.

So - I think these are equivalent.

Reply:
Let me check something and get back to you but I don't think it's equivalent.

Reply:
Okay, so using `logging.getLogger(__name__)` is what you should use in the module to get logging.  However in the script that gets run, you should use `logging.getLogger()` to get the root logger.  The problem is that `__name__` gets evaluated to `__main__` in the main script so `logger.getLogger(__name__)` there creates and configures a child of the root logger.  This means that any formatting added to the logger won't apply to logging from the module.  I've attached two files that demonstrate this.  If you run a.py, you get 

```
__name__ formatted log: WARNING testing __name__
root logger format: WARNING testing __name__
root logger format: WARNING testing root
root logger format: WARNING logging from module
```

If you comment out the second logging definition in a.py (l 9-13), you won't see the log message from b.py.

a.py:
```
import logging

logger = logging.getLogger(__name__)
formatter1 = logging.Formatter('__name__ formatted log: %(levelname)s %(message)s')
sh = logging.StreamHandler()
sh.setFormatter(formatter1)
logger.addHandler(sh)

logger2 = logging.getLogger()
formatter1 = logging.Formatter('root logger format: %(levelname)s %(message)s')
sh2 = logging.StreamHandler()
sh2.setFormatter(formatter1)
logger2.addHandler(sh2)


logger.warning("testing __name__")
logger2.warning("testing root")
import b
b.foo()
```

b.py:
```
import logging


def foo():
    logger = logging.getLogger(__name__)
    logger.addHandler(logging.NullHandler())
    logger.warning("logging from module")
```

Reply:
Ok - removed the `__name__` as this is, as you point out, the top level.